File: AppCV-Fiches/WebCV-Fiches/Models/CVViewModels/ResumeInterventionViewModel.cs
using DAL_CV_Fiches.Models.Graph;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Reflection;
using System.Threading.Tasks;

namespace WebCV_Fiches.Models.CVViewModels
{
    public class ResumeInterventionViewModel : ViewModel
    {
        public string Nombre { get; set; }
        public string Client { get; set; }
        public string Projet { get; set; }
        public int Envergure { get; set; }
        public string Fonction { get; set; }
        public string Annee { get; set; }
        public int Effors { get; set; }

        public override bool HasEdtion(EditionObject edition)
        {
 
            if(edition.ProprieteNom == "Mandats")
                return true;

 
            var propreties = MethodBase.GetCurrentMethod().DeclaringType.GetProperties().ToList();
            return propreties.Any(x => x.Name == edition.ProprieteNom);
        }
    }
}

File: AppCV-FichesFront/src/app/Models/CV/ResumeInterventionViewModel.ts
import { EditionObjecViewModel } from "../EditionObjec-model";

export class ResumeInterventionViewModel {
  nombre: number;
  entrerise: string;
  client: string;
  projet: string;
  envergure: string;
  fonction: string;
  annee: number;
  effors: number;
  debutMandat: Date;
  editionObjecViewModels: Array<EditionObjecViewModel>;
  highlight: string;
  constructor() {
    this.editionObjecViewModels = new Array<EditionObjecViewModel>();
  }
}
